test/IDE/complete_macro_declaration.swift

```diff
@@ -0,0 +1,22 @@
+// REQUIRES: swift_swift_parser
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-ide-test -batch-code-completion -source-filename %s -filecheck %raw-FileCheck -completion-output-dir %t
+
+let globalVar = 1
+macro expect(file: Int = #^DEFAULT_ARG^#) = #externalMacro(module: "MyModule", type: "MyMacro")
+// DEFAULT_ARG: Decl[GlobalVar]/CurrModule/TypeRelation[Convertible]: globalVar[#Int#]; name=globalVar
+
+@freestanding(expression)
+macro externalMacro() = ##^EXTERNAL_MACRO^#externalMacro
+// EXTERNAL_MACRO: Decl[Macro]/OtherModule[Swift]/IsSystem: externalMacro({#module: String#}, {#type: String#})[#T#]; name=externalMacro(module:type:)
+
+@freestanding(expression)
+macro externalMacroWithTrailing() = ##^EXTERNAL_MACRO_WITH_TRAILING?check=EXTERNAL_MACRO^#externalMacro
+
+@freestanding(expression)
+macro externalMacroCallPattern() = #externalMacro(#^EXTERNAL_MACRO_CALL_PATTERN^#)
+// EXTERNAL_MACRO_CALL_PATTERN: Pattern/None/Flair[ArgLabels]/TypeRelation[Convertible]: ['(']{#module: String#}, {#type: String#}[')'][#Void#]; name=module:type:
```

Comment:
What's `module:type:` there?

Reply:
That’s how the call pattern shows up in code completion, I think. It’s the same that shows up for normal function calls

```
$ cat /tmp/a.swift
func foo(module: String, type: String) {}

foo(#^COMPLETE^#)
$ swift-ide-complete /tmp/a.swift
+ /Users/alex/sbin/swift-ide-test -code-completion -source-filename /private/tmp/a.swift -code-completion-token COMPLETE
found code completion token COMPLETE at offset 47
Begin completions, 1 items
Decl[FreeFunction]/CurrModule/Flair[ArgLabels]: ['(']{#module: String#}, {#type: String#}[')'][#Void#]; name=module:type:
End completions
```